### PR TITLE
refactor logger

### PR DESCRIPTION
**Pull request by Wing924**
* improved logger performance
* make CLI output better

log sample:

file:
```
2018-12-04 01:41:20: go.d INFO: plugin[main] maximum number of used CPUs 2
2018-12-04 01:41:20: go.d INFO: plugin[main] minimum update every 1
2018-12-04 01:41:20: go.d INFO: plugin[main] loading 'tcpcheck' configuration
2018-12-04 01:41:20: go.d WARNING: plugin[main] skipping 'tcpcheck': can't find 'go.d/tcpcheck.conf' in [/etc/netdata /usr/lib/netdata/conf.d]
2018-12-04 01:41:20: go.d INFO: plugin[main] loading 'example' configuration
2018-12-04 01:41:20: go.d WARNING: plugin[main] skipping 'example': config 'jobs' section is empty or not 
2018-12-04 01:41:20: go.d INFO: plugin[main] loading 'httpcheck' configuration
2018-12-04 01:41:20: go.d WARNING: plugin[main] skipping 'httpcheck': can't find 'go.d/httpcheck.conf' in [/etc/netdata /usr/lib/netdata/conf.d]
2018-12-04 01:41:20: go.d INFO: plugin[main] start main loop
```

CLI:
```
[ INFO  ] plugin[main] logger.go:93 maximum number of used CPUs 2
[ INFO  ] plugin[main] logger.go:93 minimum update every 1
[ INFO  ] plugin[main] logger.go:93 loading 'example' configuration
[ WARN  ] plugin[main] logger.go:88 skipping 'example': config 'jobs' section is empty or not exist
[ INFO  ] plugin[main] logger.go:93 loading 'httpcheck' configuration
[ WARN  ] plugin[main] logger.go:88 skipping 'httpcheck': can't find 'go.d/httpcheck.conf' in [/etc/netdata /usr/lib/netdata/conf.d]
[ INFO  ] plugin[main] logger.go:93 loading 'tcpcheck' configuration
[ WARN  ] plugin[main] logger.go:88 skipping 'tcpcheck': can't find 'go.d/tcpcheck.conf' in [/etc/netdata /usr/lib/netdata/conf.d]
[ INFO  ] plugin[main] plugin.go:87 start main loop
[ DEBUG ] plugin[main] logger.go:98 tick 0
[ DEBUG ] plugin[main] logger.go:98 tick 1
```

### CODE REVIEW

**Comment by Wing924** [comment]
@ilyam8 it's ready. Please review it.

**Comment by ilyam8** [comment]
I haven't reviewed it but it looks nice!

File: logger/formatter.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+func (l *formatter) formatSeverity(severity Severity) {
+	if l.colored {
+		switch severity {
+		case DEBUG:
+			l.buf = append(l.buf, "\x1b[0;36m[ "...)
+		case INFO:
+			l.buf = append(l.buf, "\x1b[0;32m[ "...)
+		case WARNING:
+			l.buf = append(l.buf, "\x1b[0;33m[ "...)
```
Comment (ilyam8) [inline logger/formatter.go]: Hey, can we have human readable colors?

File: logger/logger.go [hunk 2]
```diff
@@ -106,30 +104,21 @@ func (l *Logger) print(severity Severity, a ...interface{}) {
...
 	}
-
-	l.log.Printf(
-		"go.d: %s: %s: %s: %s",
-		severity,
-		l.modName,
-		l.jobName,
-		fmt.Sprintln(a...),
-	)
+	base.log.Output(severity, l.modName, l.jobName, 3, fmt.Sprint(a...))
```
Comment (ilyam8) [inline logger/logger.go]: `l.log.Output(severity, l.modName, l.jobName, 3, fmt.Sprint(a...))`

File: logger/formatter.go (hunk 1 above)
Comment (ilyam8) [reply logger/formatter.go]: i mean in the code

Comment (Wing924) [reply logger/formatter.go]: It's highly optimized code. 3x-5x faster than previous code.

Comment (ilyam8) [reply logger/formatter.go]: simple comments will do :smile:

Comment (ilyam8) [reply logger/formatter.go]: I see 2 spaces after `INFO` and `WARN`

Comment (ilyam8) [reply logger/formatter.go]: please fix it

Comment (Wing924) [reply logger/formatter.go]: we need the space to align the log.

Comment (ilyam8) [reply logger/formatter.go]: oh, thats on purpose, i see

Comment (ilyam8) [reply logger/formatter.go]: I wonder how it looks with `CRITICAL`

Comment (ilyam8) [reply logger/formatter.go]: i think it's just `CRIT`, ok

Comment (Wing924) [reply logger/formatter.go]: in CLI, `CRITICAL` => `[ CRIT ]`

**Pull request merged**